### PR TITLE
VReplication TableStreamer: Only stream tables in tablestreamer (ignore views)

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

When we do an atomic copy the `tablestreamer` streams all tables. The corresponding `MoveTables` gets its own list of tables, for which it adds filters for each table as part of the vreplication stream's `binlogsource`. For non-atomic copy workflows, we stream each table in the `binlogsource` list of tables. On the target vreplication creates one table plan for each of these tables.

However we get a full list of tables again after we have taken a snapshot (after locking all tables) for atomic copy and stream those tables.

There was a bug where we were not excluding views from this list. Hence the target was seeing a stream from the view resulting in the target stream erroring out since it doesn't have this view in its list of table plans.

This PR fixes that bug. It also adds a view to the e2e test: that test fails without the fix in the PR.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/14648

## Checklist

-   [X] "Backport to:" labels have been added if this change should be back-ported
-   [X] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go [hunk 1]
```diff
@@ -117,12 +117,16 @@ func (ts *tableStreamer) Stream() error {
...
 	}
 
-	rs, err := conn.ExecuteFetch("show tables", -1, true)
+	rs, err := conn.ExecuteFetch("show full tables", -1, true)
 	if err != nil {
 		return err
 	}
 	for _, row := range rs.Rows {
 		tableName := row[0].ToString()
+		tableType := row[1].ToString()
```
Comment (mattlord) [inline go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go:126]: I think that we should either switch to using named rows or add a quick len(row) check just to be sure we don't crash with index out of bounds in some odd cases.

File: go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go [hunk 2]
```diff
@@ -117,12 +117,16 @@ func (ts *tableStreamer) Stream() error {
...
+		if tableType != "BASE TABLE" {
```
Comment (mattlord) [inline go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go]: We should use the existing const for this: `tmutils.TableBaseTable`

**Review by mattlord — APPROVED** [review]
Just some minor suggestions. Thanks!

File: go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go (hunk 2 above)
Comment (mattlord) [reply go/vt/vttablet/tabletserver/vstreamer/tablestreamer.go]: Whoops. You already address this. :)

**Review by shlomi-noach — APPROVED** [review]

**Pull request merged**